File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    GCD, LCM = 1, n * m
    big, small = max(n, m), min(n, m)
    
    for i in range(2, small+1):
        if not (small % i or big % i):
            GCD = i
    
    for i in range(1, small):
        isLCM = False
        for j in range(i, big):
            if small * j > big * i:
                break
            if big * i == small * j:
                LCM = big * i
                isLCM = True
        if isLCM:
            break
    
    return [GCD, LCM]

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {
    public int[] solution(int n, int m) {
        int a = n, b = m, t = b;
        
        while (a % b > 0) {
            b = a % b;
            a = t;
            t = b;
        }
        
        return new int[] {b, n * m / b};
    }
}